### PR TITLE
MED-442: pubMatic in unit integration

### PR DESCRIPTION
**Pull request by HasanSa**
- listen to the proxy new events:
1. `apester_init_inunit`:  
  - create a new inUnit ad which is `300x250` ad.
  - get event  `timeInView` as a inUnit ad display time (destroy when the timer is finished).
  - inject event `appDomain, testMode, debugLogs, bidSummaryLogs` to the `POBannerView`
2. `apester_kill_native_ad`:  destroy only the inUnit ad (a video ad from the web is playing).
- keep both bottom ad and inUnit in parallel (the bottom ad is always in front).
- implementation fixes, enhancements, cleanup and updates.

### CODE REVIEW

File: Sources/EmbededUnit/APEGADViewDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//
```
Comment (galdil) [inline Sources/EmbededUnit/APEGADViewDelegate.swift:1]: what is GAD?

File: Sources/Helpers/Constants.swift [hunk 2]
```diff
@@ -70,21 +70,29 @@ struct Constants {
...
+        static let adMob            = "adMob"
+        static let pubMatic         = "pubMatic"
+        static let profileId        = "iosProfileId"
+        static let adProvider       = "provider"
+        static let adMobUnitId      = "adUnitId"
+        static let publisherId      = "publisherId"
+        static let appStoreUrl      = "iosAppStoreUrl"
+        static let initNativeAd     = "apester_init_native_ad"
+        static let initInUnit       = "apester_init_inunit"
+        static let killInUnit       = "apester_kill_native_ad"
```
Comment (galdil) [inline Sources/Helpers/Constants.swift:87]: where do u use it?

File: Sources/EmbededUnit/APEUnitView.swift [hunk 3]
```diff
@@ -412,49 +330,186 @@ private extension APEUnitView {
...
+        
+        let delegate = pubMaticViewProviders[params.adType]?.delegate ?? makePubMaticViewDelegate(adType: params.adType)
+        pubMaticView?.delegate = delegate
+        pubMaticView?.loadAd()
+        
+        self.pubMaticViewProviders[params.adType] = PubMaticViewProvider(view: pubMaticView, delegate: delegate)
+        
+        guard  let timeInView = params.timeInView, pubMaticViewTimer == nil else { return }
+        pubMaticViewTimer = Timer.scheduledTimer(withTimeInterval: Double(timeInView), repeats: false) { _ in
+            self.hidePubMaticView()
```
Comment (galdil) [inline Sources/EmbededUnit/APEUnitView.swift:400]: does this destroy the view?
Another question, do we destroy the old views? we dont want to create a new view for each ad refresh without destroying the old one.

File: Sources/EmbededUnit/APEUnitView.swift [hunk 4]
```diff
@@ -13,28 +13,34 @@ import GoogleMobileAds
```
[the remark this answers is not in the document]
Comment (HasanSa) [reply Sources/EmbededUnit/APEUnitView.swift:13]: Done

**Pull request merged**